x-pack/plugins/infra/public/hooks/use_kibana_index_patterns.mock.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { from, of } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import { CoreStart } from '../../../../../src/core/public';
+import { FieldSpec } from '../../../../../src/plugins/data/common';
+import {
+  IIndexPattern,
+  IndexPattern,
+  IndexPatternField,
+  IndexPatternsContract,
+} from '../../../../../src/plugins/data/public';
+import { KibanaContextProvider } from '../../../../../src/plugins/kibana_react/public';
+import { Pick2 } from '../../common/utility_types';
+
+type MockIndexPattern = Pick<
+  IndexPattern,
+  'id' | 'title' | 'type' | 'getTimeField' | 'isTimeBased' | 'getFieldByName'
+>;
+export type MockIndexPatternSpec = Pick<
+  IIndexPattern,
+  'id' | 'title' | 'type' | 'timeFieldName'
+> & {
+  fields: FieldSpec[];
+};
+
+export const MockIndexPatternsKibanaContextProvider: React.FC<{
+  asyncDelay: number;
+  mockIndexPatterns: MockIndexPatternSpec[];
+}> = ({ asyncDelay, children, mockIndexPatterns }) => {
+  const indexPatterns = useMemo(
+    () =>
+      createIndexPatternsMock(
+        asyncDelay,
+        mockIndexPatterns.map(({ id, title, type = undefined, fields, timeFieldName }) => {
+          const indexPatternFields = fields.map((fieldSpec) => new IndexPatternField(fieldSpec));
+
+          return {
+            id,
+            title,
+            type,
+            getTimeField: () => indexPatternFields.find(({ name }) => name === timeFieldName),
+            isTimeBased: () => timeFieldName != null,
+            getFieldByName: (fieldName) =>
+              indexPatternFields.find(({ name }) => name === fieldName),
+          };
+        })
+      ),
+    [asyncDelay, mockIndexPatterns]
+  );
+
+  const core = useMemo<Pick2<CoreStart, 'application', 'getUrlForApp'>>(
+    () => ({
+      application: {
+        getUrlForApp: () => '',
+      },
+    }),
+    []
+  );
+
+  return (
+    <KibanaContextProvider services={{ ...core, data: { indexPatterns } }}>
+      {children}
+    </KibanaContextProvider>
+  );
+};
+
+const createIndexPatternsMock = (
```

Comment:
:information_source: Unfortunately I was unable to re-use the jest-based mocks in the storybook.